### ISSUE TITLE
[RFC] Unify appearance of `plot_param_importances` between plotly and matplotlib backends

### ISSUE THREAD

**Issue by nzw0301** [optuna/optuna#2893]
## Motivation

Plotly's `plot_param_importances` shows the importance scores explicitly as in [the optuna's docs page](https://optuna.readthedocs.io/en/stable/reference/visualization/generated/optuna.visualization.plot_param_importances.html) on the right next of each bar. On the other hand, mathplotlib's one does not show the scores as in [the optuna's docs page](https://optuna.readthedocs.io/en/stable/reference/visualization/generated/optuna.visualization.matplotlib.plot_param_importances.html) there. To minimize the difference between them, we would like to see similar plots not depending on the backends.

I think we have two options:

1. Remove importance scores from Plotly's plot
2. Add importance scores to matplotlib's plot

The reason behind the first option is that the scores mislead users sometimes, particular when the importance scores are less than `0.01`. See https://github.com/optuna/optuna/issues/2868#issue-978621430 as an example. Note that users can modify the plot object returned by plot functions in practice; users can remove them or adjust importance scores. In addition, if optuna internally removes the importance scores, it is hard for users to compute the importance score again due to its stochastic nature (see also related issue https://github.com/optuna/optuna/issues/2757).

<img width="998" alt="Screenshot 2021-09-07 at 16 11 07" src="https://user-images.githubusercontent.com/7121753/132304108-60fb0cdd-6e7e-4bdf-8033-5a2da48efa70.png">

## Description

Depending on the decision.

## Additional context (optional)

[A similar plot by scikit-learn](https://scikit-learn.org/stable/auto_examples/ensemble/plot_forest_importances.html#sphx-glr-auto-examples-ensemble-plot-forest-importances-py) does not provide such values.

**Comment by xadrianzetx** [comment]
How about dropping the label and introducing std bars similar to [sklearn](https://scikit-learn.org/stable/_images/sphx_glr_plot_forest_importances_002.png)? Would require some modifications to evaluators (`evaluate` only outputs mean importance), but generally should be possible for both fANOVA and MDI. This change could be applied both in matplotlib and plotly backends.

**Comment by nzw0301** [comment]
Thank you for your comments! Indeed, I'm not familiar with the algorithms to compute the importance scores, but it makes sense to me.

**Comment by xadrianzetx** [comment]
I would be happy to work on that, but it would be great to hear more opinions.

**Comment by nzw0301** [comment]
Let me summarise the opinions from dev-members @HideakiImamura @hvy @toshihikoyanase.

- Keep numerical values on the right next to bars. Technically we can do for the current importance score's algorithms. However, we cannot add an error bar as on sklearn's example if we introduce a deterministic importance calculation algorithm.
- Add importance scores to matplotlib's plot to unify the appearance.
- For an importance score is less than `0.01`, change the format to `<0.01`. (further opinions are welcome!)

Now, I suppose that we can resolve this issue by sending two PRs:
- change the format for importance score less than `0.01`
- add the same importance score's text to matplotlib backend as Plotly's one.

**Comment by nzw0301** [comment]
I think we can close this issue because @xadrianzetx resolves this issue by #3012 and #3073. Thank you so much for improving both `plot_param_importances` functions!
